shared/runtime/pyexec.c

```diff
@@ -143,11 +147,15 @@ static int parse_compile_execute(const void *source, mp_parse_input_kind_t input
             mp_hal_stdout_tx_strn("\x04", 1);
         }
 
-        // check for SystemExit
-        if (mp_obj_is_subclass_fast(MP_OBJ_FROM_PTR(((mp_obj_base_t *)nlr.ret_val)->type), MP_OBJ_FROM_PTR(&mp_type_SystemExit))) {
+        #ifdef MICROPY_ENABLE_VM_ABORT
```

Comment:
```suggestion
        #if MICROPY_ENABLE_VM_ABORT
```

Reply:
Also, it seems like this should be immediately after `} else {`, otherwise we are still using the MicroPython runtime (printing, etc.) and the runtime is not aborted immediately.

Reply:
It is not aborted immediately, however are those things dependent on user code preventing abort somehow? From what I understand, there are some useful things to do, e.g. flush the stdout before the abort.